gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -2015,3 +1990,41 @@ def test_{{transport_name}}_initialize_client():
     )
 
 {% endmacro %}
+
+{# rest_method_bad_request_test generates tests for rest methods
+ # which raise a google.api.core.exceptions.BadRequest error.
+#}
+{% macro rest_method_bad_request_test(service, method, method_name, transport_name, async_prefix, await_prefix, async_decorator, is_async) %}
```

Comment:
Should we add a TODO bug to follow up on making this test more generic to cover gRPC as well? For example, we could simulate a 400 response using either gRPC or REST

Reply:
addressed.